handbook/upgrade/upgrade.md

```diff
@@ -0,0 +1,280 @@
+
+!!!important "Using Virtual Machines for SystemLink Deployments"
+    It is recommended to run SystemLink Server, attached file stores, and databases within virtual machines. Virtual machines simplify the creation and restoration of backups. The use of other tools to facilitate the backup of bare metal machines, e.g.non-virtual machines, is outside the scope of this chapter.
+
+## Assumptions and Prerequisites
+
+- A server running SystemLink 2021 R1 or greater
+
+- Familiarity with installing and setting up a SystemLink Server
+
+    - [Setting up a SystemLink Server](https://www.ni.com/documentation/en/systemlink/latest/setup/)
+
+    - [Configuring NI Web Server](https://www.ni.com/documentation/en/ni-web-server/latest/manual/configuring-ni-web-server/)
+
+        - [Sign on with LDAP](/ldap/ldap)
+
+        - [Single Sign-on with OpenID Connect](/openid-connect/openid-connect/)
+
+    - Data Store Configuration
+
+        - [MongoDB](/data-stores/mongodb/)
+
+        - [Connecting to a Remote Mongo Database](https://www.ni.com/documentation/en/systemlink/latest/setup/remote-mongo-database/)
+
+        - [Connecting to a Remote File Share](https://www.ni.com/documentation/en/systemlink/latest/setup/remote-file-share/)
+
+        - [Uploading Files to Amazon Simple Storage Service (S3)](https://www.ni.com/documentation/en/systemlink/latest/data/uploading-files-to-amazon-s3/)
+
+        - [Connecting to a Remote PostgreSQL Database](https://www.ni.com/documentation/en/systemlink/latest/setup/remote-postgres-databse/)
+
+- Recommended
+
+    - All servers, file stores, and databases running in virtual machines
+
+    - Removable storage volume or device
+
+## Preparing for Upgrade or Migration
+
+Prior to attempting any upgrade or migration you should backup your server and any remote data stores. Failure to create backups increases the risk of unplanned downtime if issues occur during the upgrade or migration process.
+
+!!!note "Single Node versus Multi Node Configurations"
+    **Single node** refers to the SystemLink Server configuration where the application server, file storage, and databases used by SystemLink are all installed on the same Windows server. This is the default configuration for SystemLink Server.
+
+    **Multi node** refers to configurations where the file store or databases used by SystemLink are running on distinct hardware from the SystemLink application server. This is the recommended configuration for all production deployments.
+
+### NI-SystemLink-Migration Command Line Utility
+
+The workflows for upgrades and migrations make use of the SystemLink command line migration tool, `nislmigrate`. Refer to the readme at the [NI-SystemLink-Migration](https://github.com/ni/NI-SystemLink-Migration) GitHub repo for details on installing and using this tool.
```

Comment:
I like the idea of linking to https://pypi.org/project/nislmigrate/, since it has a link to the github repo for those who want to dig more, but is more user facing than the repo.

Reply:
Should be "make" since the subject is the plural "workflows."

Reply:
done!